cucumber/features/steps/shared.rb

```diff
@@ -8,6 +8,12 @@ def get_http_route(path)
       body = http({:method => "GET", :path => path}, {})
       response_body_to_hash(body)
     end
+
+    def switch_to_second_tab_retryable
+      begin
+        touch("UITabBarButton index:1")
+      end while query("view marked: 'Second View'").empty?
```

Comment:
I like the solution.

Let's do two things:

1.  avoid an infinite loop by putting in a counter.  Let's try 10 times.  And have a sleep of 1 second between each try.

2. Add documentation about why we are doing this.
3. Add a branch - only do this when Xcode >= 11.  I think you can also ask run loop if the Xcode is beta.  So If Xcode >= 11 && Xcode is beta.

Reply:
1) Done, but I have to set bigger timeout to make sure that build is absolutely stable.
2) Done
3) Done, I have added condition for Xcode 11 but `beta` condition doesn't work in CI since we remove `-beta` occurrence from xcode name

Reply:
Why do you remove `-beta` from the Xcode name?  That seems like a bad choice.